File: src/main/java/com/intetics/lukyanenko/config/ModelPrototypes.java
package com.intetics.lukyanenko.config;

import com.intetics.lukyanenko.models.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Scope;

@Configuration
public class ModelPrototypes {
    
    @Bean
    @Scope("prototype")
    AppUser getAppUser() {
        return new AppUser();
    }
    
    @Bean
    @Scope("prototype")
    Customer getCustomer() {
        return new Customer();
    }
    
    @Bean
    @Scope("prototype")
    GoodsCategory getGoodsCategory() {
        return new GoodsCategory();
    }
    
    @Bean
    @Scope("prototype")
    GoodsItem getGoodsItem() {
        return new GoodsItem();
    }
    
    @Bean
    @Scope("prototype")
    Order getOrder() {
        return new Order();
    }
    
    @Bean
    @Scope("prototype")
    OrderDetail getOrderDetail() {
        return new OrderDetail();
    }
}

File: src/main/java/com/intetics/lukyanenko/config/ModelPrototypes.kt
package com.intetics.lukyanenko.config

import com.intetics.lukyanenko.models.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Scope

@Configuration
open class ModelPrototypes {

    @Bean
    @Scope("prototype")
    open fun appUser(): AppUser = AppUser()

    @Bean
    @Scope("prototype")
    open fun customer(): Customer = Customer()

    @Bean
    @Scope("prototype")
    open fun goodsCategory(): GoodsCategory = GoodsCategory()

    @Bean
    @Scope("prototype")
    open fun goodsItem(): GoodsItem = GoodsItem()

    @Bean
    @Scope("prototype")
    open fun order(): Order = Order()

    @Bean
    @Scope("prototype")
    open fun orderDetail(): OrderDetail = OrderDetail()
}
